Firestore/core/src/firebase/firestore/core/transaction_runner.h

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_CORE_TRANSACTION_RUNNER_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_CORE_TRANSACTION_RUNNER_H_
+
+#include <memory>
+
+#include "Firestore/core/src/firebase/firestore/core/transaction.h"
+#include "Firestore/core/src/firebase/firestore/remote/exponential_backoff.h"
+#include "Firestore/core/src/firebase/firestore/remote/remote_store.h"
+#include "Firestore/core/src/firebase/firestore/util/async_queue.h"
+#include "Firestore/core/src/firebase/firestore/util/status.h"
+
+namespace core = firebase::firestore::core;
+namespace util = firebase::firestore::util;
+namespace remote = firebase::firestore::remote;
+
+namespace firebase {
+namespace firestore {
+namespace core {
+
+const int kRetryCount = 5;
```

Comment:
I believe this declares a `firebase::firestore::core::kRetryCount` variable that will be visible to any code that includes this `.h` file...  but we don't want/need it to be that widely visible.  So this should probably be `static const int kRetryCount = 5;` in the .mm file instead.

Reply:
I see you made it static, but I think we additionally want to move it to the .mm file...  or actually, we could make it a private member of the TransactionRunner class, similar to what we do with TargetIdGenerator::kReservedBits [here](https://github.com/firebase/firebase-ios-sdk/blob/1553801fec6ebd93804b56b701ad0b529bf49de3/Firestore/core/src/firebase/firestore/core/target_id_generator.h#L90).

Basically we want this constant to be scoped to the TransactionRunner code rather than being a "global" `firebase::firestore::core::kRetryCount` constant that's exposed to all client code.  If you move it to a `static const` in the .mm file then that's scoping it to just that one file.  If you move it to a private member in the TransactionRunner class, then that's scoping it to just the members of the TransactionRunner class.  Either works, though the latter is probably cleaner on second thought.